contracts/Staking.sol

```diff
@@ -216,6 +216,9 @@ contract Staking is Pauser, Whitelist {
         uint256 shares = _tokenToShare(_tokens, validator.tokens, validator.shares);
 
         dt.Delegator storage delegator = validator.delegators[delAddr];
+        if (delegator.shares == 0) {
```

Comment:
I am not quite sure of the detail, but if the delegator is added to the validator's delAddrs, while line 222 adds additional shares to the delegator, would it cause an issue?

Reply:
Line 219 is to tell if the delegator is new to the validator.  Line 220 only adds the delegator address to the list when it does not exist. I think delegator.shares += shares should work correctly when delegator.shares == 0, right?

Reply:
What if shares == 0 and delegator.shares == 0?  This means that we may add delAddr twice?

Reply:
I mean it is possible `shares == 0`?  Looks like it is likely impossible, but we may need a `require(shares != 0, "...")` to avoid the case.

Reply:
Agree.